### PR TITLE
Enable quantum farmer

### PR DESCRIPTION
**Pull request by Hamms**
After all that work to make the quantum nature of our apps generalizable, I completely forgot to put the necessary functionality where it needs to be to actually take advantage of that and enable quantum farmer.

quantum farmer
image

Formerly, Maze would have a copy of its map, which it uses when doing Farmer or regular Maze stuff. But Bee had its own copy of the map and ignored Maze's map because Bee had to do more complicated stuff.

This moves most of that complicated functionality into MazeMap, so that Bee can just use Maze's map again rather than keeping its own.

It also moves some functionality that was formerly in BeeCell into its parent class (Cell).

### CODE REVIEW

File: apps/src/maze/bee.js [hunk 1]
```diff
@@ -27,112 +27,20 @@ var Bee = function (maze, studioApp, config) {
...
-      return row.map(function (mapCell, y) {
-        var initialDirtCell = config.level.initialDirt[x][y];
-        return BeeCell.parseFromOldValues(mapCell, initialDirtCell);
-      });
-    });
-  }
-  this.staticGrids = Bee.getAllStaticGrids(this.variableGrid);
-
-  this.currentStaticGridId = 0;
-  this.currentStaticGrid = this.staticGrids[0];
```
Comment (Hamms) [inline apps/src/maze/bee.js:61]: this has all been moved to MazeMap

File: apps/src/maze/bee.js [hunk 2]
```diff
@@ -27,112 +27,20 @@ var Bee = function (maze, studioApp, config) {
...
- */
-Bee.prototype.resetCurrentValues = function () {
-  this.currentStaticGrid.forEach(function (row) {
-    row.forEach(function (cell) {
-      cell.resetCurrentValue();
-    });
-  });
-};
-
-/**
```
Comment (Hamms) [inline apps/src/maze/bee.js]: this has (mostly) all been moved to Cell

Comment (Hamms) [inline apps/src/maze/bee.js:126]: this has (mostly) all been moved to MazeMap

**Comment by islemaster** [comment]
Awesome! LGTM.

**Pull request merged**